addons/project_timesheet_holidays/models/hr_holidays.py

```diff
@@ -139,3 +139,12 @@ def _action_user_cancel(self, reason):
         timesheets.unlink()
         self._check_missing_global_leave_timesheets()
         return res
+
+    def _force_cancel(self, *args, **kwargs):
+        """
+        override this method to reevaluate timesheets after the leaves are updated via force cancel
+        """
+        super()._force_cancel(*args, **kwargs)
+        timesheets = self.sudo().timesheet_ids
+        timesheets.write({'holiday_id': False})
```

Comment:
```suggestion
        timesheets.holiday_id = False
```
looks better, but it is a detail